cartography/graph/statement.py

```diff
@@ -67,22 +74,25 @@ def _run(self, session):
 
     def _run_iterative(self, session):
         """
-        Iterative statement execution.
-
-        Expects the query to return the total number of records updated.
+        Runs the statement in batches of `LIMIT_SIZE` until `TotalCompleted` returns 0.
+        This follows large delete transaction best practices in Neo4j:
+        https://neo4j.com/developer/kb/large-delete-transaction-best-practices-in-neo4j/
         """
         self.parameters["LIMIT_SIZE"] = self.iterationsize
-
-        # TODO improve this
-        done = False
-        while not done:
-            results = self._run(session)
-            done = False
-            for r in results:
-                if int(r['TotalCompleted']) == 0:
-                    done = True
-
-                break
+        self._run_iter_core(session)
+
+    @tenacity.retry(
```

Comment:
One problem is that if the Neo4j statement starts before the timeout and if a deadlock happens during the Neo4j statement, I'm not sure if this decorator has the ability to cancel the statement itself. I've tested the `backoff` package but it doesn't solve this problem; maybe `tenacity` does.

Reply:
This PR looks good, but just wondering if it would be possible to do this without adding an external dependency?

Reply:
@marco-lancini In my testing, this change doesn't seem to fix the issue I was hoping :\. Probably will let this PR go stale. Do you have the same #434 issue?

Reply:
Interesting, never happened to our deployment

Reply:
Good to know that it's probably not a code issue then.
I'm on Neo4j 3.5.19 enterprise, python neo4j driver 1.7.6, Python 3.6.2. How about you?

Reply:
Neo4j 3.5.11